sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -832,7 +833,10 @@ private[hive] class HiveClientImpl(
     val (partCols, schema) = table.schema.map(toHiveColumn).partition { c =>
       table.partitionColumnNames.contains(c.getName)
     }
-    if (schema.isEmpty) {
+
+    // after SPARK-19279, it is not allowed to create a hive table with an empty schema,
```

Comment:
Uh... I found a bug in my original code for [inferSchema](https://github.com/gatorsmile/spark/blob/f7e0264c6786fb69d313406beba066300a63f3b1/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala#L583). Let me fix it now. The schema could be not empty when it is a partitioned table.